tests/collections/tts/modules/test_transformer_2501.py

```diff
@@ -0,0 +1,164 @@
+# Copyright (c) 2025, NVIDIA CORPORATION & AFFILIATES.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+import random
+
+import numpy as np
+import pytest
+import torch
+
+from nemo.collections.tts.modules.transformer_2501 import Transformer
+
+
+def set_seed(seed):
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    np.random.seed(seed)
+    random.seed(seed)
+
+
+@pytest.mark.unit
+class TestTransformer:
+    @classmethod
+    def setup_class(cls):
+        cls.n_layers = 1
+        cls.d_model = 4
+        cls.d_ffn = 16
+        cls.sa_n_heads = 2
+        cls.kernel_size = 3
+        cls.p_dropout = 0.0
+        cls.p_dropout_out = 0.0
+        cls.has_xattn = False
+        cls.is_causal = True
+        cls.max_length_causal_mask = 10
+
+        # fmt:off
+        cls.input_tensor = torch.Tensor(
+            [[[ 0.7049,  0.0305, -0.8542,  0.5388],
+              [-0.5265, -1.3320,  1.5451,  0.4086],
+              [-2.0546,  0.5259,  0.5995, -0.4078],
+              [ 0.4530, -0.3918,  2.1403, -0.2062],
+              [-0.0984,  0.4855,  0.7076,  0.0431],
+              [-0.4394, -0.6761,  1.7389, -0.9423],
+              [ 0.9764, -1.0889, -0.1634,  2.2799],
+              [ 0.2277,  0.0367,  0.3680,  0.9759],
+              [ 0.8760,  1.4248, -0.2724,  0.9353],
+              [-1.4920, -0.5683, -0.9277,  2.1160]]]
+        )
+        # fmt:on
+
+    def test_forward_causal_self_attn_and_no_xattn(self):
+        set_seed(0)
+        model = Transformer(
+            n_layers=self.n_layers,
+            d_model=self.d_model,
+            d_ffn=self.d_ffn,
+            sa_n_heads=self.sa_n_heads,
+            kernel_size=self.kernel_size,
+            p_dropout=self.p_dropout,
+            p_dropout_out=self.p_dropout_out,
+            has_xattn=self.has_xattn,
+            is_causal=self.is_causal,
+            max_length_causal_mask=self.max_length_causal_mask,
+        )
+
+        # Check model init
+        assert torch.isclose(torch.mean(model.layers[0].pos_ff.proj.weight), 0.0)
+        assert torch.isclose(torch.std(model.layers[0].pos_ff.proj.weight), 0.02)
+        assert torch.isclose(torch.mean(model.layers[0].pos_ff.o_net.weight), 0.0)
+        assert torch.isclose(torch.std(model.layers[0].pos_ff.o_net.weight), 0.02 / math.sqrt(2.0))
```

Comment:
bug: *** AttributeError: 'ConvolutionLayer' object has no attribute 'weight'.

```suggestion
        assert torch.isclose(torch.mean(model.layers[0].pos_ff.proj.conv.weight), 0.0)
        assert torch.isclose(torch.std(model.layers[0].pos_ff.proj.conv.weight), 0.02)
        assert torch.isclose(torch.mean(model.layers[0].pos_ff.o_net.conv.weight), 0.0)
        assert torch.isclose(torch.std(model.layers[0].pos_ff.o_net.conv.weight), 0.02 / math.sqrt(2.0))
```

Reply:
fixed.